### ISSUE TITLE
[Enhancement] Condense To(Not(a)) calls to single ToNot(a)

### ISSUE THREAD

**Issue by sbrownjc** [nunnatsa/ginkgolinter#205]
**Is your feature request related to a problem? Please describe.**
Gomega has `To()` and `Not()` functions that can be nested like `To(Not())` and `Not(To())`, however it also has `ToNot()` and `NotTo()` functions that perform the same, also applicable to `Should()`.

See https://pkg.go.dev/github.com/onsi/gomega#Assertion

**Describe the solution you'd like**
A rule/matcher to detect nested To/Not functions and suggest the single function as a replacement.

**Describe alternatives you've considered**

Using gofmt's rewrite rules, either plain or configured in golangci-lint, does not appear to be effective. Example golangci-lint config:

```yaml
version: "2"
formatters:
  enable:
    - gofmt
  settings:
    gofmt:
      rewrite-rules:
        - pattern: 'To(Not(a))'
          replacement: 'ToNot(a)'
        - pattern: 'Not(To(a))'
          replacement: 'NotTo(a)'
        - pattern: 'Should(Not(a))'
          replacement: 'ShouldNot(a)'
        - pattern: 'Not(Should(a))'
          replacement: 'ShouldNot(a)'
```

**Comment by nunnatsa** [comment]
Thanks for the idea, @sbrownjc .

I started looking at it. I wonder if we want it enable by default, and if yes, we want to allow disabling the new rule. Maybe someone can find it more readable?
